### ISSUE TITLE
incorrect transformation (pose)

### ISSUE THREAD

**Issue by gebawe** [johnwlambert/waymo_to_argoverse#17]
Hello @johnwlambert 

Thank you for this great work.

I have checked the conversion script and it seems the pose extracted from the waymo has some problem.

I have tried to transform multiple point clouds into the city coordinate frame and it seems there is a bug (see fig, the trees are moving). Can you please check this? thank you

![image](https://user-images.githubusercontent.com/30930452/148538471-d3244814-58d7-4cd1-b2ed-ef6f25e36dba.png)

**Comment by johnwlambert** [comment]
Hi @awethaileslassie, thanks for the heads up. Which log id is this from? Did you run
```bash
python waymo_raw_data_to_argoverse.py
```
(can you post your full command here)?

**Comment by gebawe** [comment]
> Hi @awethaileslassie, thanks for the heads up. Which log id is this from? Did you run

WOD training dataset
log id: 10072140764565668044_4060_000_4080_000

> ```shell
> python waymo_raw_data_to_argoverse.py
```
Yes I have run this and got the dataset converted into argoverse (eg. files)
![image](https://user-images.githubusercontent.com/30930452/148562257-819ae4a6-fc7d-442e-a9f6-c7f73b5bcfeb.png)

 
> (can you post your full command here)?

I am using the argoverse-api coordinate transformation, to transform point clouds from multiple frames into city coordinate system

lidar_pts_p = city_to_egovehicle_se3.transform_point_cloud(lidar_pts[:,:3])  # put into city coords
			
However, the static objects such as buildings and trees appear as moving objects.

I have tried my coordinate transformation in the original argoverse dataset and it is working well (see fig).
![image](https://user-images.githubusercontent.com/30930452/148563352-fe53c5a6-ce66-48e2-9219-59b5bd70cad6.png)

**Comment by gebawe** [comment]
@johnwlambert 

Yes I have run this and got the dataset converted into argoverse (eg. files)
![image](https://user-images.githubusercontent.com/30930452/148562257-819ae4a6-fc7d-442e-a9f6-c7f73b5bcfeb.png)

 
> (can you post your full command here)?

I am using the argoverse-api coordinate transformation, to transform point clouds from multiple frames into a city coordinate system

lidar_pts_p = city_to_egovehicle_se3.transform_point_cloud(lidar_pts[:,:3])  # put into city coords
			
However, the static objects such as buildings and trees appear as moving objects.

I have tried my coordinate transformation in the original argoverse dataset and it is working well (see fig).
![image](https://user-images.githubusercontent.com/30930452/148563352-fe53c5a6-ce66-48e2-9219-59b5bd70cad6.png)

**Comment by johnwlambert** [comment]
Thanks @awethaileslassie. Unfortunately I don't have time to debug this much this week, but I'm sure it's an easy fix.

If you have time, one suggestion would be: take the first two successive point clouds from this log, and try to estimate the translation between them (suppose rotation is small/none if the car is driving straight). Then we can compare with the relative pose as suggested by the dumped poses. It may be that the Waymo Open Dataset is actually storing the inverse of the pose, or something of that nature. Could you share what you find here?

**Comment by gebawe** [comment]
@johnwlambert 

Sure, I have tried to debug the problem and it seems it is related to the rotmat2quat function.
I am directly using the 4x4 transformation matrix directly from the WOD dataset and now I am getting the correct transformed PCL (see fig).

![image](https://user-images.githubusercontent.com/30930452/148772996-9d506fbb-aa3f-4f10-9892-501e96f5b4c3.png)
